sklearn/cross_decomposition/pls_.py

```diff
@@ -271,6 +271,10 @@ def fit(self, X, Y):
 
         # NIPALS algo: outer loop, over components
         for k in range(self.n_components):
+            if np.all(np.dot(Yk.T, Yk) < np.finfo(np.double).eps):
+                # Yk constant
+                warnings.warn('Y constant at iteration %s' % k)
+                break
```

Comment:
For reference, if this is commented out we get the following kind of error when running the common tests:

```
======================================================================
ERROR: sklearn.tests.test_common.test_non_meta_estimators('PLSCanonical', <class 'sklearn.cross_decomposition.pls_.PLSCanonical'>)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/volatile/ogrisel/envs/py35/lib/python3.5/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/volatile/ogrisel/code/scikit-learn/sklearn/utils/estimator_checks.py", line 407, in check_estimators_nan_inf
    estimator.fit(X_train_finite, y)
  File "/volatile/ogrisel/code/scikit-learn/sklearn/cross_decomposition/pls_.py", line 334, in fit
    linalg.pinv(np.dot(self.x_loadings_.T, self.x_weights_)))
  File "/volatile/ogrisel/envs/py35/lib/python3.5/site-packages/scipy/linalg/basic.py", line 602, in pinv
    a = np.asarray_chkfinite(a)
  File "/volatile/ogrisel/envs/py35/lib/python3.5/site-packages/numpy/lib/function_base.py", line 613, in asarray_chkfinite
    "array must not contain infs or NaNs")
ValueError: array must not contain infs or NaNs
```